pkg/controller/queuejob/queuejob_controller_ex.go

```diff
@@ -581,6 +584,48 @@ func (qjm *XController) GetAggregatedResourcesPerGenericItem(cqj *arbv1.AppWrapp
 
 	return retVal
 }
+
+//Gets all objects owned by AW from API server, check user supplied status and set whole AW status
+func (qjm *XController) GetAllObjectsOwned(caw *arbv1.AppWrapper) arbv1.AppWrapperState {
+
+	// Get all pods and related resources
+	countCompletedItems := 0
```

Comment:
Suggest changing to a bool: `  completionStatusFound = false`.   See below for more details.

Reply:
I am not sure how will changing to boolean help, we want to :
- Get all items that have completed status
- Get the status of all the items that get completed as they execute
- Get the count of items that are required to the completed by the user and items that are marked completed by controllers, if the count of those two matches, then set the status as Complete or RunningHoldCompletion.

Reply:
The change to boolean is referenced below for determining if there was at least 1 item where `competionstatus`  was set in order to ensure the state change does not happen for AWs that do not set `competionstatus`  in any AW item.

Is it necessary to go through each item if, while iterating through each item, a `competionstatus` is not met?  For example: if there a 6 items in the AW, but only item 1 and item 6 have `competionstatus` defined to ["Completed", "Failed"], if the first item (item 1) is still "Running" is there a need to continue looking at the rest of the items since the condition to change the state is not met?  Can the function return immediately without checking items 2-6?  If the loop completes then all items all items without `competionstatus` have been ignored and all items with `competionstatus` conditions have been met.  Conditions for changing the AW state have been met with a final check to see if there was at least one item with `competionstatus` (the suggested boolean).

Reply:
ok thanks, I think I now understand, I am doing early termination now, please review

Reply:
Awesome!

Reply:
lgtm!